University of Wisconsin-Madison Libraries Catalog.js

```diff
@@ -0,0 +1,256 @@
+			default:
+				return "book";
+		}
+	} 
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('a.item_path[href*="/catalog/"]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	 
+	var risURL = url.replace(/[#?].*$/, '') + '.ris';
+	//there is also a link tag for RIS but the href is currently missing
+
+	ZU.doGet(risURL, function(text) {
+		//delete birth/death year from author name
+		text = text.replace(/^(AU\s+-.*), \d\d\d\d-(\d\d\d\d)?$/m, "$1");
+
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+		translator.setString(text);
+		translator.translate();
+	})
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://search.library.wisc.edu/catalog/9910104568102121",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Zotero : a guide for librarians, researchers, and educators",
+				"creators": [
+					{
+						"lastName": "Puckett",
+						"firstName": "Jason",
+						"creatorType": "author"
+					}
+				],
+				"date": "2011",
+				"abstractNote": "xi, 159 pages : illustrations ; 23 cm",
+				"callNumber": "PN171.F56 P83 2011",
+				"libraryCatalog": "University of Wisconsin-Madison Libraries Catalog",
+				"publisher": "Chicago : Association of College and Research Libraries, 2011.",
+				"shortTitle": "Zotero",
+				"url": "https://search.library.wisc.edu/catalog/9910104568102121",
+				"attachments": [],
+				"tags": [],
+				"notes": [
+					{
+						"note": "<p>Includes bibliographical references.</p>"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://search.library.wisc.edu/catalog/9912334246702121",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Fiscal year 2017 budget amendments : communication from the President of the United States transmitting FY 2017 budget amendments for national security activities at the Department of Defense, the Department of State, and the U.S. Agency for International Development to fund overseas contingency operations",
+				"creators": [
+					{
+						"lastName": "United States. President (2009-2017 : Obama)",
+						"creatorType": "author",
+						"fieldMode": 1
+					}
+				],
+				"date": "2016",
+				"abstractNote": "38 pages ; 24 cm",
```

Comment:
It looks like they map the MARC/PNX field for size/extent (or whatever it's called) here. a) I don't think this makes much sense in the abstract field and b) could we try to find /(\d+)\s*pages/[1] at least for books and move it there?

Reply:
I think it is called extent and dimension, see https://www.loc.gov/marc/bibliographic/bd300.html . Yes, this looks like an error that the physical description is mapped to the `AB` field in RIS.

I will delete these things from the abstract and move the number of pages to `numPages` field. But I will use another regexp. Please have a look.